profiling/src/profiling/thread_utils.rs

```diff
@@ -69,3 +75,67 @@ pub fn join_timeout(handle: JoinHandle<()>, timeout: Duration, impact: &str) {
         std::panic::resume_unwind(err)
     }
 }
+
+thread_local! {
+    static THREAD_NAME: OnceCell<String> = const { OnceCell::new() };
+}
+
+pub fn get_current_thread_name() -> String {
+    THREAD_NAME.with(|name| {
+        name.get_or_init(|| {
+            #[cfg(php_zts)]
+            let mut thread_name = SAPI.to_string();
+            #[cfg(not(php_zts))]
+            let thread_name = SAPI.to_string();
+
+            #[cfg(php_zts)]
+            {
+                // So far, only FrankenPHP sets meaningful thread names
+                if *SAPI == Sapi::FrankenPHP {
+                    let mut name = [0u8; 32];
+
+                    let result = unsafe {
+                        libc::pthread_getname_np(
+                            libc::pthread_self(),
+                            name.as_mut_ptr() as *mut c_char,
+                            name.len(),
+                        )
+                    };
+
+                    if result == 0 {
+                        // If successful, convert the result to a Rust String
+                        let cstr =
+                            unsafe { std::ffi::CStr::from_ptr(name.as_ptr() as *const c_char) };
+                        let str_slice: &str = cstr.to_str().unwrap();
```

Comment:
This will crash if it's not utf-8. I think we should log instead and use the SAPI name only if this happens. What do you think?

Reply:
I changed the `unwrap()` to a `unwrap_or_default()` (which makes it empty). So in that case we'd just have the SAPI name and nothing else which is okay I'd say